### PR TITLE
Issue #626 Fix MyPy errors

### LINKED ISSUE #681: Move the GroundWaterFlow a GrounWaterTransport models to separate files
In GitLab by @Manangka on Dec 7, 2023, 1:36

### PR DESCRIPTION
**Pull request by Manangka**
This also fixes #681, #682, #684

### CODE REVIEW

File: imod/mf6/package.py [hunk 1]
```diff
@@ -40,8 +40,9 @@ class Package(PackageBase, abc.ABC):
     """
 
     _pkg_id = ""
-    _init_schemata = {}
```
Comment (JoerivanEngelen) [inline imod/mf6/package.py:43]: I don't think ``ValueSchema`` is the right type here, as that is quite a specific implementation. I think we need to define a ``SchemaType`` or something for this.

**Review by JoerivanEngelen — CHANGES_REQUESTED** [review]
This cleans up things nicely and better type annotation is good. I have one comment on the type annotation used for schemata.

File: imod/mf6/model.py [hunk 2]
```diff
@@ -501,6 +531,10 @@ def _get_regridding_domain(
                 included_in_all = regridded_idomain
             else:
                 included_in_all = included_in_all.where(regridded_idomain.notnull())
+
+        if included_in_all is None:
+            raise ValueError("Unable to regrid")
```
Comment (luitjansl) [inline imod/mf6/model.py:536]: pleas expand the error message a bit to explain why is is unable to regrid

File: imod/util.py [hunk 3]
```diff
@@ -1159,7 +1159,7 @@ def _replace(
     )
 
 
-def values_within_range(da, min=None, max=None):
+def values_within_range(da, min: Optional[float] = None, max: Optional[float] = None):
```
Comment (luitjansl) [inline imod/util.py:1162]: return type missing

**Review by luitjansl — APPROVED** [review]
looks good

File: imod/mf6/package.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/mf6/package.py:44]: I means defining something like this in ``schemata.py``

```python
from typing import TypeVar

SchemaType = TypeVar("SchemaType", bound="BaseSchema")
```

Comment (Manangka) [reply imod/mf6/package.py:44]: Applied this

**Review by JoerivanEngelen — APPROVED** [review]
Looks good! Bit of a shocker that we forgot to add ``fastcore`` to the ``pyproject.toml`` :-O

**Pull request merged**